src/pytorch_lightning/utilities/apply_func.py

```diff
@@ -13,327 +13,70 @@
 # limitations under the License.
 """Utilities used for collections."""
 
-import dataclasses
-from abc import ABC
-from collections import defaultdict, OrderedDict
-from copy import deepcopy
-from functools import partial
-from typing import Any, Callable, List, Mapping, Optional, Sequence, Tuple, Union
-
-import numpy as np
-import torch
-from torch import Tensor
-
-from pytorch_lightning.utilities.exceptions import MisconfigurationException
-
-_BLOCKING_DEVICE_TYPES = ("cpu", "mps")
-
-
-def to_dtype_tensor(
-    value: Union[int, float, List[Union[int, float]]], dtype: torch.dtype, device: Union[str, torch.device]
-) -> Tensor:
-    return torch.tensor(value, dtype=dtype, device=device)
-
-
-def from_numpy(value: np.ndarray, device: Union[str, torch.device]) -> Tensor:
-    return torch.from_numpy(value).to(device)
-
-
-CONVERSION_DTYPES: List[Tuple[Any, Callable[[Any, Any], Tensor]]] = [
-    # bool -> uint8 as bool -> torch.bool triggers RuntimeError: Unsupported data type for NCCL process group
-    (bool, partial(to_dtype_tensor, dtype=torch.uint8)),
-    (int, partial(to_dtype_tensor, dtype=torch.int)),
-    (float, partial(to_dtype_tensor, dtype=torch.float)),
-    (np.ndarray, from_numpy),
-]
```

Comment:
Just noticed we didn't deprecate this. But it's very tricky to do, being a global list.

Should we try or yolo?

Reply:
It's going to far imo :)